src/Components/OrderHistory.js

```diff
@@ -0,0 +1,163 @@
+import { UserContext } from "../services/UserContext";
+
+import moment from 'moment';
+
+import {Tabs, Tab, TabContainer, TabContent, TabPane, ButtonGroup} from 'react-bootstrap/'
+import {Button,Card,Container,Row,Col,Form, Dropdown, DropdownButton, Modal, ModalDialog, ModalBody, ListGroup, ListGroupItem} from 'react-bootstrap'; // eslint-disable-line
+
+import AuthService from '../services/auth.service';
+
+const OrderHistory = () => {
+    const { user, login, logout } = useContext(UserContext);
+    const [orderHistory, setOrderHistory] = useState([])
+    const [listingHistory, setListingHistory] = useState([])
+
+
+    const [listings, setListing] = useState([])
+
+    const [isLoaded, setIsLoaded] = useState(false);
+    const [isPageLoaded, setIsPageLoaded] = useState(false); 
+
+    const [isReady, setReady] = useState(false)
+
+
+    const fetchOptions = {
+        headers: new Headers({'Authorization':'bearer '+ AuthService.getCurrentUser()})
+    }
+
+    const approveOrder = (id,status) => {
+        fetch(`https://gariunaicloud.azurewebsites.net/api/Orders/${id}/status?status=${status}&force=True`, {
+            method: 'POST',
+            headers: {
+                'Authorization':'bearer '+ AuthService.getCurrentUser()
+            }
+        }).then(response => {
+            return response.json();
+        })
+        .then(data => {
+            fetchOrderHistory("listingHistory")
+        })
+    }
+    
+    const fetchOrderHistory = async (which) => {
+        const types = {
+            orderHistory: 'https://gariunaicloud.azurewebsites.net/api/Orders/placed',
+            listingHistory: 'https://gariunaicloud.azurewebsites.net/api/Orders/received'
+        };
+        fetch(types[which], fetchOptions)
+          .then(response => {
+            return response.json()
+          })
+          .then(data => {
+            data.forEach(async element => {
+                await fetch("https://gariunaicloud.azurewebsites.net/api/Listings/"+element.listingId)
+                    .then(response => {
+                        return response.json()
+                    })
+                    .then(data => {
+                        element['listing'] = data
+                    })
+              })
+              setTimeout(() => {
+                  if(which == "orderHistory"){
+                      setOrderHistory(data)
+                      console.log(data)
+                      setReady(true)
+                  }else{
+                      setListingHistory(data)
+                      console.log(data)
+                      setReady(true)
+                  }
+              }, 600);
+            
+          })
+      } 
+    
+    const handleHistory= async (which)=>{
+        await fetchOrderHistory(which)
+      }
+    
+    useEffect(() => {
+        if(!isPageLoaded){
+            setIsLoaded(true);
+        }
+    }, []);
+
+    useEffect(() => {
+
+    }, [listingHistory])
```

Comment:
what's the point of this, just to reload page on listingHistory change?

Reply:
Yes